tests/python/pytorch/graphbolt/impl/test_neighbor_sampler.py

```diff
@@ -41,8 +41,12 @@ def get_hetero_graph():
 @unittest.skipIf(F._default_context_str != "gpu", reason="Enabled only on GPU.")
 @pytest.mark.parametrize("hetero", [False, True])
 @pytest.mark.parametrize("prob_name", [None, "weight", "mask"])
-def test_NeighborSampler_GraphFetch(hetero, prob_name):
-    items = torch.arange(3)
+@pytest.mark.parametrize("sorted", [False, True])
+def test_NeighborSampler_GraphFetch(hetero, prob_name, sorted):
+    if sorted:
+        items = torch.arange(3)
+    else:
+        items = torch.tensor([2, 0, 1])
```

Comment:
I am a little bit confused by the unit test case, does it suggest that regardless whether the items are sorted or not, the result will be the same?

Reply:
Yes, however, the code path may be a little different. Sorting is required when using hetero graphs because the extracted InSubgraph needs to have the node_type_offset field. When nodes are not sorted, it is not possible to assign a value to that field.

Reply:
Whether the nodes are sorted or not, the output should match with the nonpipelined version.